src/notation/view/partlistmodel.cpp

```diff
@@ -279,14 +279,18 @@ void PartListModel::openSelectedParts()
         return;
     }
 
+    std::sort(rows.begin(), rows.end());
+
     ExcerptNotationList newExcerpts;
     for (int index : rows) {
         newExcerpts.push_back(m_excerpts[index]);
     }
 
     masterNotation()->addExcerpts(newExcerpts);
 
-    context()->setCurrentNotation(m_excerpts[rows.last()]->notation());
+    for (int index : rows) {
+        context()->setCurrentNotation(m_excerpts[i]->notation());
```

Comment:
Better to use INotation::setOpened here:

 for (int index : rows) {
 m_excerpts[i]->notation()->setOpened(true);
 }
 
setCurrentNotation should be used only once (after that cycle with the last notation in the selected list) because it is quite expensive:
 
 context()->setCurrentNotation(m_excerpts[rows.last()]->notation());

Reply:
Ok will do that. Thanks!